IBPSA/Fluid/Geothermal/Aquifer/MultiWell.mo

```diff
@@ -0,0 +1,419 @@
+        rotation=-90,
+        origin={-80,50})));
+  Airflow.Multizone.Point_m_flow powHot(
+    redeclare final package Medium = Medium,
+    m=1,
+    final dpMea_nominal=dp_nominal_aquifer,
+    final mMea_flow_nominal=m_flow_nominal)
+                "Pressure drop in the warm side of the aquifer" annotation (
+      Placement(transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=-90,
+        origin={80,-10})));
+  FixedResistances.PressureDrop resHot(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final dp_nominal=dp_nominal_well)
+    "Pressure drop in the warm well" annotation (
+      Placement(transformation(
+        extent={{10,10},{-10,-10}},
+        rotation=-90,
+        origin={80,50})));
+  Movers.Preconfigured.SpeedControlled_y movCol(
+    redeclare final package Medium = Medium,
+    addPowerToMedium=false,
+    final m_flow_nominal=m_flow_nominal,
+    final dp_nominal=dp_nominal_aquifer*2 + dp_nominal_well*2 + dp_nominal_hex)
+    "Pump to extract from cold well" annotation (Placement(transformation(
+        extent={{-10,10},{10,-10}},
+        rotation=90,
+        origin={-80,20})));
+  Movers.Preconfigured.SpeedControlled_y movHot(
+    redeclare final package Medium = Medium,
+    addPowerToMedium=false,
+    final m_flow_nominal=m_flow_nominal,
+    final dp_nominal=dp_nominal_aquifer*2 + dp_nominal_well*2 + dp_nominal_hex)
+    "Pump to extract from hot well" annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={80,20})));
+protected
+  Modelica.Blocks.Nonlinear.Limiter limiter(
+    final uMax=1,
+    final uMin=0)
+    "Limiter for pump signal"
+    annotation (Placement(transformation(extent={{-60,40},{-40,60}})));
+  Modelica.Blocks.Math.Gain gain(
+    final k=-1) "Inversion of control signal"
+    annotation (Placement(transformation(extent={{-8,40},{12,60}})));
+  Modelica.Blocks.Nonlinear.Limiter limiter1(
+    final uMax=1,
+    final uMin=0)
+    "Limiter for pump signal"
+    annotation (Placement(transformation(extent={{30,40},{50,60}})));
+  parameter Modelica.Units.SI.Radius r[nVol + 1](each fixed=false)
+    "Radius to the boundary of the i-th domain";
+  parameter Modelica.Units.SI.Radius rC[nVol](each fixed=false)
+    "Radius to the center of the i-th domain";
+  parameter Modelica.Units.SI.SpecificHeatCapacity cpWat(fixed=false)
+    "Water specific heat capacity";
+  parameter Modelica.Units.SI.Density rhoWat(fixed=false)
+    "Water density";
+  parameter Modelica.Units.SI.ThermalConductivity kWat(fixed=false)
+    "Water thermal conductivity";
+  parameter Modelica.Units.SI.HeatCapacity C[nVol](each fixed=false)
+    "Heat capacity of segment";
+  parameter Modelica.Units.SI.Volume VWat[nVol](each fixed=false)
+    "Volumes of water";
+  parameter Modelica.Units.SI.ThermalResistance R[nVol+1](each fixed=false)
+    "Thermal resistances between nodes";
```

Comment:
`R` has dimension `nVol+1`, but above it is used in
```
  Modelica.Thermal.HeatTransfer.Components.ThermalResistor theResCoo[nVol](
    R=R/nCoo)
```
The dimension must match (OpenModelica catches this mistake).

Reply:
It should be ok now